### PR TITLE
refactor(calendar): set time for dates created by calendar to midnight

### PR DESCRIPTION
**Pull request by mhchen**
@jelbourn Please review at your convenience.

### CODE REVIEW

File: src/components/calendar/dateUtil.js [hunk 1]
```diff
@@ -184,7 +185,26 @@
...
+     * Creates a date with the time set to midnight.
+     * Drop-in replacement for two forms of the Date constructor:
+     * 1. No argument for Date representing now.
+     * 2. Single-argument value representing number of seconds since Unix Epoch.
+     * @param {number=} opt_value
+     * @return {Date} New date with time set to midnight.
+     */
+    function createDateAtMidnight(opt_value) {
+      var date;
+      if (typeof opt_value !== 'undefined') {
```
Comment (jelbourn) [inline src/components/calendar/dateUtil.js]: You can use `angular.isUndefined`
https://docs.angularjs.org/api/ng/function/angular.isUndefined

File: src/components/calendar/calendar.js [hunk 2]
```diff
@@ -220,7 +220,7 @@
     this.buildWeekHeader();
     this.hideVerticalScrollbar();
 
-    this.displayDate = this.selectedDate || new Date(Date.now());
+    this.displayDate = this.selectedDate || this.dateUtil.createDateAtMidnight();
```
Comment (jelbourn) [inline src/components/calendar/calendar.js]: This could actually be 
```js
  this.displayDate = this.selectedDate || this.today;
```

**Comment by jelbourn** [comment]
LGTM aside from those two comments. Thanks for taking care of this!

**Comment by mhchen** [comment]
@jelbourn No problem, comments addressed!

**Comment by jelbourn** [comment]
Closed via 7124d644d80a06757e4af37a1151d21f3ce316da

**Pull request closed without merging**